### ISSUE TITLE
Location-specific dance roles limit

### ISSUE THREAD

**Issue by afs2015** [django-danceschool/django-danceschool#142]
We have a location that has a registration limit of 32 people and wanted to limit it to 16 leads and 16 follows registering.

We can currently do that at the event level but it would nice to be able to restrict this at the location level as well so we don't have to do it every time we create a class series.

**Comment by leetucker** [comment]
@afs2015 I agree that this makes sense, and I can try to add it soon.  The logical order of precedence should be:

event-specified > location-specified > default limit.

**Comment by leetucker** [comment]
@afs2015 OK, so now that I'm looking at the code, I think that it is in fact designed to do more or less what you want already.  However, there is a design flaw/bug that is preventing it from working quite as intended.

Take a look at the `capacityForRole()` method in the Event model.  If no custom role limits are specified, then the capacity for each role is set as the value of the total event capacity divided by the number of roles.  The total event capacity is set automatically in the `save()` method of Event, and if no capacity is specified, then it uses the capacity specified for the Location or Room. `capacityForRole()` is what is used in the `soldOutForRole()` method to determine whether registration for a specific role is allowed.  So, a class with a capacity of 32 and two roles 'Lead' and 'Follow' should automatically allow 16 registrations for each role.

The issue is the in the admin, when you create a new Series, the capacity field is auto-populated with the default capacity for the school.  That effectively negates the logic in Event.save(), since capacity is always specified for the Event in the admin.  What needs to happen is that when you select or change a Location for the Series, it should update the capacity *unless* the capacity has already been updated by the user.  So, I think what we need is:

1. Locations and Rooms have their capacity passed to the admin via a data- attribute (I thought this was already in place but I guess my memory is mistaken)
2. A Javascript flag keeps track of whether the user has edited the capacity field.
3. If the user has not edited the capacity field, then when the location is changed, fill in capacity using the data-capacity attribute on the Location.

I agree that this needs to be fixed, but since there are a number of steps I'm going to deal with some higher-priority things first.  If you think that the logic for capacities should work differently, let me know and I'm happy to consider alternatives.
